ams/server/src/main/java/com/netease/arctic/server/optimizing/TaskRuntime.java

```diff
@@ -128,8 +132,6 @@ public void ack(OptimizerThread thread) {
         () -> {
           validThread(thread);
           statusMachine.accept(Status.ACKED);
```

Comment:
Literally speaking, would it be more appropriate to set threadId and token in the ack stage? Of course schedule can also get `threadId`.

Reply:
The `SCHEDULED` tasks also should have the correct token and threadId, when checking suspending tasks, this token will be used. Here is the related code `com.netease.arctic.server.DefaultOptimizingService.OptimizerKeeper#buildSuspendingPredication`
```java
    private Predicate<TaskRuntime> buildSuspendingPredication(
        String token, boolean isOptimizerExpired) {
      return task -> {
        if (isOptimizerExpired) {
          return token.equals(task.getToken()); // Here, tasks with the same token with expired optimizer should retry
        } else {
          return token.equals(task.getToken())
              && task.getStatus() == TaskRuntime.Status.SCHEDULED
              && task.getStartTime() + taskAckTimeout < System.currentTimeMillis();
        }
      };
    }
```

Reply:
I see, igonre.